Formula/rbspy.rb

```diff
@@ -0,0 +1,17 @@
+class Rbspy < Formula
+  desc "Sampling profiler for Ruby"
+  homepage "https://rbspy.github.io/"
+  url "https://github.com/rbspy/rbspy/archive/v0.3.2.tar.gz"
+  sha256 "8250692165937e1060ab3c94f6d4742658873073ca08ee50f6d546c7b84807ff"
+
+  depends_on "rust" => :build
+
+  def install
+    system "cargo", "install", "--root", prefix, "--path", "."
+  end
+
+  test do
+    output = shell_output("#{bin}/rbspy -V")
```

Comment:
Any chance of a test that exercises a deeper level of functionality? Thanks!

Reply:
Hi Dom, thanks for the feedback.  

I've thought about this for a bit.  Rbspy's snapshot and record commands record stack traces from a running process but needs to use sudo to do so on MacOS (https://rbspy.github.io/using-rbspy/).  Would we want a command that requires sudo to be used in a formula test?

I could write a test to use the report command (this allows you to show a report based on the record command without sudo), but that requires a gzipped file containing the results of a previously run record.

Let me know what you think we should do in this case.

Reply:
CI can't use `sudo` and it's not something we encourage/support in general, so that's a no-go for better or worse.

> I could write a test to use the report command (creates report based on the output of the record command) which does not require sudo. However, this requires a gzipped file containing the results of a previously run record.

This sounds promising. Any chance that's something upstream provides as part of an examples directory or is that something we'd have to create/store/etc ourselves?

Reply:
It's something we would have to create in the formula.  
1. Create a sample recording 
```
rbspy01
{"trace":[{"name":"<c function>","relative_path":"unknown","absolute_path":null,"lineno":0},{"name":"aaa","relative_path":"short_program.rb","absolute_path":"/Users/home/brew/sample_program.rb","lineno":2},{"name":"bbb","relative_path":"sample_program.rb","absolute_path":"/Users/home/brew/sample_program.rb","lineno":6},{"name":"ccc","relative_path":"sample_program.rb","absolute_path":"/Users/home/brew/sample_program.rb","lineno":10},{"name":"<main>","relative_path":"sample_program.rb","absolute_path":"/Users/home/brew/sample_program.rb","lineno":13}],"pid":38387}
```
2. gzip the contents
3. Run the report on the gzipped contents which outputs a report file.  
4. Verify the report file has certain text, for instance, `100.00   100.00  aaa - short_program.rb`

If you think that's OK, I can try adding this to the formula.

Reply:
Alternatively, a sample recording file could be added upstream and used in a unit test.  If that was done, we could use that for the formula test.  @jvnas what do you think?

Reply:
Alternatively, a sample recording file could be added upstream and used in a unit test.  If that was done, we could use that for the formula test.  @jvns what do you think?

Reply:
I'd merge a PR that adds a sample recording file.

Reply:
I'm happy with either approach. If we write it in the formula you'll need to wrap the creation of the file to 80 characters using `<<~EOS` and then use `.delete!` to stitch it back together again (see `spigot` for an example). Everything else sounds easily achievable and something we're happy for a test to do.

Reply:
Thanks for the wrapping suggestions, they were very helpful.  Since you're OK with putting it in the formula I just added it in there.  I realized that we can base64 the gzip file too.